### PR TITLE
Add 8.0.0-rc2 release notes

### PR DESCRIPTION
**Pull request by pugnascotia**
Preview: https://elasticsearch_83412.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/8.0/release-notes-8.0.0-rc2.html

### CODE REVIEW

**Comment by pugnascotia** [comment]
@elasticmachine run elasticsearch-ci/bwc 
@elasticmachine run elasticsearch-ci/part-2

**Comment by pugnascotia** [comment]
@elasticmachine test this please

**Comment by lockewritesdocs** [comment]
I'm wondering if we're missing some release notes that should be in both `rc1` and `rc2`. For example, #80554 is listed as a deprecation under the ML section in the [8.0.0-rc1 release notes](https://www.elastic.co/guide/en/elasticsearch/reference/8.0/release-notes-8.0.0-rc1.html#deprecation-8.0.0-rc1). That PR has the `v8.0.0-rc2` label, but isn't included in the 8.0.0-rc2 release notes -- should it (and others like it) be included?

**Comment by pugnascotia** [comment]
@lockewritesdocs I excluded that PR deliberately because it's a forward-port of #80545, and was actually released in 7.16.0.

**Comment by lockewritesdocs** [comment]
> @lockewritesdocs I excluded that PR deliberately because it's a forward-port of #80545, and was actually released in 7.16.0.

Got it! Thanks for the explanation @pugnascotia.

File: docs/reference/release-notes/8.0.0-rc2.asciidoc [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+Cluster Coordination::
+* Make `TaskBatcher` less lock-heavy {es-pull}82227[#82227] (issue: {es-issue}77466[#77466])
+
+ILM+SLM::
+* Avoid unnecessary `LifecycleExecutionState` recalculation {es-pull}81558[#81558] (issues: {es-issue}77466[#77466], {es-issue}79692[#79692])
+* Make unchanged ILM policy updates into no-op {es-pull}82240[#82240] (issue: {es-issue}82065[#82065])
+
+Infra/Core::
+* Prevent upgrades to 8.0 from 7.non-last {es-pull}82321[#82321] (issue: {es-issue}81865[#81865])
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/8.0.0-rc2.asciidoc:52]:
```suggestion
* Prevent upgrades to 8.0 without first upgrading to the last 7.x release {es-pull}82321[#82321] (issue: {es-issue}81865[#81865])
```

File: docs/reference/release-notes/8.0.0-rc2.asciidoc [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+Machine Learning::
+* Add `deployment_stats` to trained model stats {es-pull}80531[#80531]
+* Setting `use_auto_machine_memory_percent` now defaults to `max_model_memory_limit` {es-pull}80532[#80532] (issue: {es-issue}80415[#80415])
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/8.0.0-rc2.asciidoc:56]:
```suggestion
* The setting `use_auto_machine_memory_percent` now defaults to `max_model_memory_limit` {es-pull}80532[#80532] (issue: {es-issue}80415[#80415])
```

File: docs/reference/release-notes/8.0.0-rc2.asciidoc [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+Network::
+* Improve slow inbound Handling to include response type {es-pull}80425[#80425]
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/8.0.0-rc2.asciidoc:59]:
```suggestion
* Improve slow inbound handling to include response type {es-pull}80425[#80425]
```

File: docs/reference/release-notes/8.0.0-rc2.asciidoc [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+* Use adjusted total memory instead of total memory {es-pull}80528[#80528] (issue: {es-issue}78750[#78750])
+
+Infra/Scripting::
+* Fix duplicated allow lists upon script engine creation {es-pull}82820[#82820] (issue: {es-issue}82778[#82778])
+
+Ingest::
+* Adjust default geoip logging to be less verbose {es-pull}81404[#81404] (issue: {es-issue}81356[#81356])
+
+Machine Learning::
+* Check `total_definition_length` is consistent on start deployment {es-pull}80553[#80553]
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/8.0.0-rc2.asciidoc:79]:
```suggestion
* Check that `total_definition_length` is consistent before starting a deployment {es-pull}80553[#80553]
```

File: docs/reference/release-notes/8.0.0-rc2.asciidoc [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+* Fail inference processor more consistently on certain error types {es-pull}81475[#81475]
+* Optimize the job stats call to do fewer searches {es-pull}82362[#82362] (issue: {es-issue}82255[#82255])
+
+Recovery::
+* Make shard started response handling only return after the CS update completes {es-pull}82790[#82790] (issue: {es-issue}81628[#81628])
```
Comment (lockewritesdocs) [inline docs/reference/release-notes/8.0.0-rc2.asciidoc:84]:
```suggestion
* Make shard started response handling only return after the cluster state update completes {es-pull}82790[#82790] (issue: {es-issue}81628[#81628])
```

File: docs/reference/migration/migrate_8_0.asciidoc [hunk 6]
```diff
@@ -105,5 +105,33 @@ user password is already configured.
...
+[[deprecate-max-merge-at-once-explicit-setting]]
+.The setting `max_merge_at_once_explicit` is deprecated.
+[%collapsible]
+====
+*Details* +
+The setting `max_merge_at_once_explicit` is deprecated as this setting has
+been removed in Lucene 9.
+
+*Impact* +
+You should remove the setting `max_merge_at_once_explicit` from your
```
Comment (lockewritesdocs) [inline docs/reference/migration/migrate_8_0.asciidoc:118]:
```suggestion
Remove the setting `max_merge_at_once_explicit` from your
```

File: docs/reference/migration/migrate_8_0.asciidoc [hunk 7]
```diff
@@ -105,5 +105,33 @@ user password is already configured.
...
+.The setting `indices.query.bool.max_clause_count` is deprecated.
+[%collapsible]
+====
+*Details* +
+The setting `indices.query.bool.max_clause_count` is deprecated. Instead,
+{es} configures the maximum clause count for Lucene based on the available
+heap and the size of the thread pool.
+
+*Impact* +
+You should remove the setting `indices.query.bool.max_clause_count` from
```
Comment (lockewritesdocs) [inline docs/reference/migration/migrate_8_0.asciidoc:132]:
```suggestion
Remove the setting `indices.query.bool.max_clause_count` from
```

**Review by lockewritesdocs — APPROVED** [review]
Added a few non-blocking grammatical changes and suggestions. Thanks @pugnascotia!

**Pull request merged**